Sources/swift-format/main.swift

```diff
@@ -48,6 +50,41 @@ fileprivate func main(_ arguments: [String]) -> Int32 {
   }
 }
 
+/// Load the configuration.
+private func loadConfiguration(
+  forSwiftFile swiftFilePath: String, configFilePath: String?
+) -> Configuration {
+  if let path = configFilePath {
+    return decodedConfiguration(fromFileAtPath: path)
+  }
+  else {
+    // Search for a ".swift-format" configuration file in the directory of the current .swift file,
+    // or its nearest parent.
+    let swiftFileDir = URL(fileURLWithPath: swiftFilePath).deletingLastPathComponent()
+    return decodedConfiguration(
+      fromFileAtPath: findConfigurationFile(inDirectory: swiftFileDir.path))
+  }
+}
+
+/// Look for a ".swift-format" configuration file in "inDirectory", or its nearest parent. Return
+/// "nil" if one is not found.
+private func findConfigurationFile(inDirectory: String) -> String? {
+  let cwd = FileManager.default.currentDirectoryPath
+  var path = URL(
+    fileURLWithPath: AbsolutePath(inDirectory, relativeTo: AbsolutePath(cwd)).asString)
+  let configFilename = ".swift-format"
+
+  while path.path != "/.." {
+    let testPath = path.appendingPathComponent(configFilename).path
+    if FileManager.default.fileExists(atPath: testPath) {
```

Comment:
Does `fileExists` return true if the file exists but isn't readable by the process's UID? We should probably check that as well.

Reply:
Maybe use `isReadableFile` instead?

Reply:
Yeah, that sounds like exactly what we need.